functions/src/api/authentificateJWT.ts

```diff
@@ -0,0 +1,55 @@
+import * as jsonWebToken from 'jsonwebtoken'
+import { Request, Response } from 'express'
+import * as functions from 'firebase-functions'
+import { VerifyErrors } from 'jsonwebtoken'
+import { JWTData } from '../types/JWTData'
+import { JWTToken } from '../types/JWTToken'
+import { isTokenValidInDb } from '../db/isTokenValidInDb'
+
+export const authenticateJWT = (
+    req: Request,
+    res: Response,
+    next: Function
+) => {
+    const { jwt } = functions.config()
+    if (!jwt.key) {
+        throw new functions.https.HttpsError(
+            'failed-precondition',
+            'Missing JWT Key'
+        )
+    }
+
+    const authHeader = req.headers.authorization
+
+    if (authHeader) {
+        const token = <JWTToken>authHeader.split(' ')[1]
+
+        jsonWebToken.verify(
+            token,
+            jwt.key,
+            {
+                algorithms: ['HS256'],
+            },
+            async (err: VerifyErrors | null, data: {} | undefined) => {
+                if (err) {
+                    return res.sendStatus(403)
+                }
+                if (!data) {
+                    return res.sendStatus(403)
+                }
+                const jwtData: JWTData = <JWTData>data
+
+                const isTokenValid = await isTokenValidInDb(jwtData, token)
+
+                if (!isTokenValid) {
+                    return res.sendStatus(403)
```

Comment:
although the status are well known, I'd prefer to read explicit names instead of numbers for the status codes:
`403` -> `statusCode.FORBIDDEN` ?

Reply:
https://fr.wikipedia.org/wiki/Liste_des_codes_HTTP 
:D 
un peu de par coeur même si les deux son valide :D

Reply:
:P I know them as well :D just consider the explicit words more readable and elegant :) but it's just an opinion, it's ok to me if you prefer to keep the numbers